addon/scripts/tokens.js

```diff
@@ -23,16 +23,27 @@
 //  r blinding factor, sjcl bignum
 function CreateBlindToken() {
     let t = newRandomPoint();
+    if (!t.point) {
+        return null;
+    }
     let bpt = blindPoint(t.point);
     return { token: t.token, point: bpt.point, blind: bpt.blind };
 }
 
 // returns: array of blind tokens
 function GenerateNewTokens(n) {
+    let tokens = [];
     let i = 0;
-    let tokens = new Array(n);
-    for (i = 0; i < tokens.length; i++) {
-        tokens[i] = CreateBlindToken();
+    while (i < n) {
```

Comment:
seems to me using `` for (var i=0; i<n; i++)`` makes more sense here

Reply:
You're right this is a weird way of doing it. I wrote it this way to prevent empty array entries. However I've rewritten it so that the array is populated and then empty entries are removed afterwards.